### PR TITLE
fix(elasticsearch sink):  Elasticsearch sink with api_version set to "auto" does not recognize the API version of ES6 as V6 (#17226)

### PR DESCRIPTION
**Pull request by syedriko**
fix(elasticsearch sink):  Elasticsearch sink with api_version set to "auto" does not recognize the API version of ES6 as V6 (#17226)

### CODE REVIEW

File: src/sinks/elasticsearch/common.rs [hunk 1]
```diff
@@ -276,29 +276,35 @@ async fn get_version(
     tls_settings: &TlsSettings,
     proxy_config: &ProxyConfig,
 ) -> crate::Result<usize> {
+    #[derive(Deserialize)]
+    struct Version {
+        number: Option<String>,
+    }
     #[derive(Deserialize)]
     struct ClusterState {
```
Comment (spencergilbert) [inline src/sinks/elasticsearch/common.rs:284]: This seems incorrectly named given the changes, I'm also not sure if we need an additional struct or just the `version.number`?

**Review by spencergilbert — COMMENTED** [review]
Thank you this contribution! Definitely seems like we're missing test coverage on this as well, so thank you for surfacing that as well 🙂 

It _looks_ like we could cut out one of the structs, or at least rename it. What do you think @syedriko?

File: src/sinks/elasticsearch/common.rs (hunk 1 above)
Comment (bruceg) [reply src/sinks/elasticsearch/common.rs:284]: I think this is right, at least the structure, since the version number is nested this way.

Comment (spencergilbert) [reply src/sinks/elasticsearch/common.rs:284]: Mhmm, yea - I don't see a name for that API, so maybe `Response` is fine?

Comment (syedriko) [reply src/sinks/elasticsearch/common.rs]: Thanks for the feedback! `response` was already taken so I renamed ClusterState to ResponsePayload. Couldn't find the name of that API either. If there's a more concise way to express the shape of that payload for serde, I'm all for it. I also added a debug!() log statement for the auto-detected API version.

File: src/sinks/elasticsearch/common.rs [hunk 2]
```diff
@@ -145,7 +145,10 @@ impl ElasticsearchCommon {
                     )
                     .await
                     {
-                        Ok(version) => version,
+                        Ok(version) => {
+                            debug!(message = "Auto-detected Elasticsearch API version:", %version);
```
Comment (spencergilbert) [inline src/sinks/elasticsearch/common.rs:149]: 👍 good call! I think it'll be caught by a check we have on log messages.

```suggestion
                            debug!(message = "Auto-detected Elasticsearch API version.", %version);
```

File: src/sinks/elasticsearch/common.rs [hunk 3]
```diff
@@ -145,7 +145,10 @@ impl ElasticsearchCommon {
...
+                            version
+                        },
```
Comment (spencergilbert) [inline src/sinks/elasticsearch/common.rs:151]:
```suggestion
                        }
```

`rustfmt`

**Review by spencergilbert — CHANGES_REQUESTED** [review]
Awesome, thank you! Two suggestions to get CI green.

**Review by spencergilbert — APPROVED** [review]
Thanks for the fix!

**Pull request merged**